Забелин Михаил Михайлович ИУ3-32Б

Hacker1470 opened the pull request.

Hacker1470 committed:
Update README.md

Hacker1470 committed:
Initial commit

Hacker1470 committed:
fix: small style changes

IU3Labs commented on lab/lab1/src/solutions/Task0.java:
```diff
@@ -0,0 +1,98 @@
+        else return -1;
+    }
+
+    public void start() {
```
Сначала public методы, потом все остальные по нарастанию скрытия. Сначала методы вызова, потом те, которые вызываются в порядке вызова.

Привести в порядок.

IU3Labs replied:
переменные то зачем уехали.
Сначала поля класса, потом методы.

IU3Labs commented on lab/lab1/src/solutions/Task0.java:
```diff
@@ -0,0 +1,98 @@
+        scanner = externalScanner;
+    }
+
+    private int getLengthFromConsole() {
```
Методы повторяются из класса в класс. Нарушает DRY

Решение. Отдельный класс для работы с массивом.

IU3Labs commented on lab/lab1/src/solutions/TaskB4.java:
```diff
@@ -0,0 +1,109 @@
+        return result;
+    }
+
+    private void printBooleanArray(boolean[] booleans) {
```
boolean без особых проблем в консоль выводится true и false, делать проверки  if (booleans[i]) = изобретать велосипед.

Hacker1470 committed:
fix: following DRY and style edits

- Added new static class ComsolePresets for repeating methods.
- Changed position of methods to follow Java-style coding

Hacker1470 committed:
fix: new names for class and methods

ConsolePresets -> ConsoleArrayPresets:
* getLengthFromConsole -> getLength
* getNumberArrayFromConsole -> getNumberArray

Hacker1470 commented:
Исправлено

IU3Labs reviewed (approved):
Комментарий на будущее
